internal/node/node.bzl

```diff
@@ -130,19 +131,23 @@ def _nodejs_binary_impl(ctx):
     node_modules = depset(ctx.files.node_modules)
 
     # Also include files from npm fine grained deps as inputs.
-    # These deps are identified by the NodeModuleSources provider.
+    # These deps are identified by the NpmPackageInfo provider.
     for d in ctx.attr.data:
-        if NodeModuleSources in d:
-            node_modules = depset(transitive = [node_modules, d[NodeModuleSources].sources])
+        if NpmPackageInfo in d:
+            node_modules = depset(transitive = [node_modules, d[NpmPackageInfo].sources])
 
     # Using a depset will allow us to avoid flattening files and sources
     # inside this loop. This should reduce the performances hits,
     # since we don't need to call .to_list()
     sources = depset()
 
     for d in ctx.attr.data:
-        if hasattr(d, "node_sources"):
-            sources = depset(transitive = [sources, d.node_sources])
+        if DeclarationInfo in d:
+            sources = depset(transitive = [sources, d[DeclarationInfo].transitive_declarations])
```

Comment:
why do we care about declarations in a nodejs_binary?

Reply:
good catch. thanks. removed.